### PR TITLE
docs: update application development content

### PR DESCRIPTION
**Pull request by eoaksnes**
## What does this pull request change?

* docs: update application development content

### CODE REVIEW

File: docs/docs/guides/application-development.md [hunk 1]
```diff
@@ -53,58 +48,129 @@ my-app/
...
 To run dependent services locally you can use the included `docker-compose.yaml` file. This file includes databases, DMSS service, and job service. 
 
-Start services by running:
+Start services in the background (-d detached mode) by running:
 
-```
-docker compose up
+```bash
+docker-compose pull
+docker compose up --build -d
```
Comment (soofstad) [inline docs/docs/guides/application-development.md]:
```suggestion
docker compose up
```
There is nothing to build in this compose file. Every image is pulled.
Also, perhaps dont recommend running detached? Most people running locally will be interested in viewing the logs.

File: docs/docs/guides/application-development.md [hunk 2]
```diff
@@ -53,58 +48,129 @@ my-app/
...
+We recommend create a virtual Python environment. 
+
+Create a new virtual environment by running: 
+
+```bash
+python3 -m venv .venv
+``` 
 
-To populate DMSS, run:
+Activate the virtual environment by running:
```
Comment (soofstad) [inline docs/docs/guides/application-development.md:106]: Perhaps remove all this python/venv stuff?
Will propably not apply to many environments (like windows in Equinor) anyway.
Just shorten in to "Install the python package dm-cli"

File: docs/docs/guides/application-development.md [hunk 3]
```diff
@@ -53,58 +48,129 @@ my-app/
...
+When you’re ready to deploy to production, create a minified bundle by running the command:
 
+```bash
+npm run build
+```
 
-See [domain modeling](./domain-modeling.md) for how to create and add models for the app.
+## Configure the app
+
+This template uses an application entity to control the behavior of the app. This entity can be seen at `app/data/DemoApplicationDataSource/instances/demoApplication.json`. The `.env` points to this application entity, which will be loaded and used to select what UI plugin to be shown. The application recipe link that defines what to show is located at `app/data/DemoApplicationDataSource/DemoApplication/recipes/demoApp.json`.
```
Comment (soofstad) [inline docs/docs/guides/application-development.md]:
```suggestion
This template uses an application entity to control the behavior of the app. This entity can be seen at `app/data/DemoApplicationDataSource/instances/demoApplication.json`. The `.env`-file has a variable pointing to this application entity, which will be loaded and used to select what _UIPlugin_ to be shown. The application _RecipeLink_ that defines what to show is located at `app/data/DemoApplicationDataSource/DemoApplication/recipes/demoApp.json`. 
```

File: docs/docs/guides/application-development.md [hunk 4]
```diff
@@ -53,58 +48,129 @@ my-app/
...
+
+To ensure consistent code formatting and to do a minimum of checks on the local computer before committing code to a Git repository, the app provides a `.pre-commit-config.yaml` file that is used to setup Git _pre-commit hooks_.
+The pre-commit hook will run [prettier](https://prettier.io/) formatting, certain tests, sanitisers and analyze the JavaScript code using [eslint](https://eslint.org/).
+To run pre-commit, it needs to be installed on your local machine with
+
+```bash
+pip install pre-commit
+```
+
+This tell pre-commit to always run for this repository on every commit.
```
Comment (soofstad) [inline docs/docs/guides/application-development.md:165]: "pre-commit install" is missing here, if you want it to run automatically

**Review by soofstad — APPROVED** [review]

File: docs/docs/guides/application-development.md (hunk 2 above)
Comment (eoaksnes) [reply docs/docs/guides/application-development.md:106]: I made it a tip (in a accordion), so that it is not that visible 👍

**Pull request merged**